perllib/FixMyStreet/Script/Merton/SendWaste.pm

```diff
@@ -54,4 +55,50 @@ sub _problems {
     });
 }
 
+sub send_comments {
+    my ($self) = @_;
+    my $comments = $self->_comments;
+
+    while (my $row = $comments->next) {
+        my $problem = $row->problem;
+
+        my $cobrand = $problem->get_cobrand_logged;
+        FixMyStreet::DB->schema->cobrand($cobrand);
+        $cobrand->set_lang_and_domain($problem->lang, 1);
+
+        my $cfg = $cobrand->feature('echo');
+        my %o311_cfg = (
+            jurisdiction => '',
+            endpoint => $cfg->{open311_endpoint},
+            api_key => $cfg->{open311_api_key},
+            extended_statuses => $self->body->send_extended_statuses,
+            fixmystreet_body => $self->body,
+            use_customer_reference => 1,
+        );
+        my $o = Open311->new(%o311_cfg);
+        $problem->set_extra_metadata( customer_reference => $problem->get_extra_metadata('crimson_external_id') );
+        my $id = $o->post_service_request_update( $row );
+        if ( $id ) {
+            $row->set_extra_metadata( sent_to_crimson => 1 );
+            $row->set_extra_metadata( crimson_external_id => $id );
+            $row->update;
+        } else {
+            STDERR->print("Failed to post over Open311\n\n" . $o->error . "\n");
+        }
+    }
+}
+
+sub _comments {
+    my $self = shift;
+    FixMyStreet::DB->resultset('Comment')->to_body($self->body->id)->search({
+        'problem.cobrand_data' => 'waste',
+        'problem.cobrand' => 'merton',
+        'problem.extra' => { '\?' => 'sent_to_crimson' },
+        'me.external_id' => \'is not null',
+        -not => { 'me.extra' => { '\?' => 'sent_to_crimson' } },
```

Comment:
As extra can be null, this will fail when it is, because the inverse of null is still null. See Bromley.pm for an example, you want something like `-or => [ 'me.extra' => undef, -not => { 'me.extra' => { '\?' => 'sent_to_crimson' } } ],`. Should probably make it have an empty default or try and hide that wrinkle somewhere perhaps.

Reply:
Aha, thanks. Have updated and added a couple more tests to cover more cases.